lib/ladybug/energy_model/shade.rb

```diff
@@ -0,0 +1,88 @@
+# specific prior written permission from the respective party.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'ladybug/energy_model/model_object'
+
+require 'json-schema'
+require 'json'
+require 'openstudio'
+
+module Ladybug
+  module EnergyModel
+    class Shade < ModelObject
+      attr_reader :errors, :warnings
+
+      def initialize(hash)
+        super(hash)
+        raise "Incorrect model type '#{@type}'" unless @type == 'Shade'
+      end
+
+      def defaults
+        result = {}
+        result
+      end
+
+      def find_existing_openstudio_object(openstudio_model)
+        object = openstudio_model.getSurfaceByName(@hash[:name])
+        return object.get if object.is_initialized
+        nil
+      end
+
+      def create_openstudio_object(openstudio_model)
+        openstudio_vertices = OpenStudio::Point3dVector.new
+        @hash[:geometry][:boundary].each do |vertex|
+          openstudio_vertices << OpenStudio::Point3d.new(vertex[0], vertex[1], vertex[2])
+        end
+
+        openstudio_construction = nil
+        if @hash[:properties][:energy][:construction]
+          construction_name = @hash[:properties][:energy][:construction]
+          construction = openstudio_model.getConstructionByName(construction_name)
+          unless construction.empty?
+            openstudio_construction = construction.get
+          end
+        end
+
+        openstudio_shading_surface = OpenStudio::Model::ShadingSurface.new(openstudio_vertices, openstudio_model)
+        openstudio_shading_surface.setName(@hash[:name])
+        openstudio_shading_surface.setConstruction(openstudio_construction) if openstudio_construction
+        openstudio_shading_surface.setTransmittanceSchedule(@hash[:transmittance_schedule]) if @hash[:transmittance_schedule]
+        
+        openstudio_shading_surface_group = OpenStudio::Model::ShadingSurfaceGroup.new(openstudio_model)
+        
+        openstudio_shading_surface.setShadingSurfaceGroup(openstudio_shading_surface_group)
```

Comment:
Hm. Does this mean that every since Shade surface get's its own `ShadingSurfaceGroup`?  In that case, what is the purpose of the group and is there a way that we can just not use it? Of can we use it how it is supposed to be used?

Reply:
@DavidGoldwasser what is the purpose of the ShadingSurfaceGroup for each surface?